cloudinit/cmd/main.py

```diff
@@ -478,9 +544,22 @@ def main_init(name, args):
         # dhcp clients to advertize this hostname to any DDNS services
         # LP: #1746455.
         _maybe_set_hostname(init, stage="local", retry_stage="network")
+
     init.apply_network_config(bring_up=bring_up_interfaces)
 
     if mode == sources.DSMODE_LOCAL:
+        if _should_wait_on_network(init.datasource):
+            LOG.debug(
+                "Network connectivity determined necessary for "
+                "cloud-init's network stage"
```

Comment:
"Future me" in triage wants to know the reason we chose to wait on network. Can we log what condition triggered our network wait: no datasource vs non-#cloud-config user-data vs write_files in config.  I also want this in logs to make it conspicuous why we are waiting in the event that we shouldn't be in some cases.